docs/src/BrandingCssVarsProvider.tsx

```diff
@@ -77,7 +77,7 @@ export default function BrandingCssVarsProvider(props: { children: React.ReactNo
   const { children } = props;
   return (
     // need to use deprecated API because MUI X repo still on Material UI v5
-    <ThemeVarsProvider theme={theme} disableTransitionOnChange>
+    <ThemeVarsProvider theme={theme} disableTransitionOnChange forceRecalculateTheme>
```

Comment:
This will be removed in #45386. Have to do this so that the docs does not break.

Reply:
Maybe `forceThemeRecalculation` is better. The prop is part of the ThemeProvider, so maybe we can even call it `forceRecalculation`. This is an advanced prop i.e. users need to take a look at the docs to use it properly, so a shorter name is fine for me.

Q: Is "recalculation" the right term/concept? Would "rerender" be better? Would `forceRerender` make sense?

Reply:
I think rerender is a side-effect that cause by a new theme. I still think that it should have the word `theme` as part of the prop.

What's about `forceThemeRewrite` or `forceThemeRerender`?

Reply:
`forceThemeRerender` sounds good to me. wdyt @DiegoAndai?

Reply:
Adding a new option 😅: `forceThemeRecreate`, what do you think?

The issue with `forceThemeRerender` is that we're recreating the theme, which is not a component, so there's no concept of rendering. The rerender is a side effect of the recreated theme,

Reply:
Recreating the theme (or not) sounds like an implementation detail, while rerendering is what users care about.

No strong opinion, in either case users would certainly need to learn what this prop does from the docs.

Reply:
> Sounds like an implementation detail, while rerendering is what users care about.

You're right, this is the angle we should look at it. I'm on board with `forceThemeRerender`

Reply:
Updated to `forceThemeRerender`. Ready for final review.